synapse_data_warehouse/synapse/dynamic_tables/V2.33.4__file_latest_dynamic_table.sql

```diff
@@ -0,0 +1,24 @@
+-- Introduce the dynamic table
+USE SCHEMA {{database_name}}.synapse; --noqa: JJ01,PRS,TMP
+CREATE OR REPLACE DYNAMIC TABLE FILE_LATEST
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS 
+    WITH dedup_filesnapshots AS (
+        SELECT
+            *
+        FROM {{database_name}}.SYNAPSE_RAW.FILESNAPSHOTS --noqa: TMP
+        WHERE
+            SNAPSHOT_DATE >= CURRENT_TIMESTAMP - INTERVAL '30 days'
+        QUALIFY
+            ROW_NUMBER() OVER (
+                PARTITION BY ID
+                ORDER BY CHANGE_TIMESTAMP DESC, SNAPSHOT_TIMESTAMP DESC
+            ) = 1
+    )
+    SELECT 
+        * 
+    FROM 
+        dedup_filesnapshots
+    WHERE
+        STATUS = 'AVAILABLE' OR NOT IS_PREVIEW;
```

Comment:
One comment, we currently provide analytics on the distribution of file handle statuses.  I think these would be great "gold layer" filters, but since we don't provide the raw tables to people, I think providing some of this info in the silver layer would still be valuable for some.

Most file statistic queries are done with joins against the node_latest table which makes it more rare to get a STATUS='UNLINKED' or IS_PREVIEW unless you query this table alone.

I think consulting platform about whether filehandles can be deleted will be valuable here.

Reply:
Agreed. We don't want to be filtering data beyond deduplication at this point. What if someone wants a dashboard that tells them how much data has been garbage-collected (STATUS = "ARCHIVED") or how much storage we currently dedicate to file previews (IS_PREVIEW = TRUE)? We don't want to have to build a dashboard from the bronze-level table.